--- code/blockp
+++ SHRDLU/BLOCKP
@@ -1,11 +1,11 @@
 
-################################################################
-
-		BLOCKP >
-
-
-    MICRO-PLANNER CODE FOR THE "BLOCKS" MICRO-WORLD 
-################################################################
+;;;################################################################
+;;;
+;;;		BLOCKP >
+;;;
+;;;
+;;;    MICRO-PLANNER CODE FOR THE "BLOCKS" MICRO-WORLD 
+;;;################################################################
 
 
 (DEFPROP TA-AT
@@ -242,7 +242,7 @@
 					     (TCENT (QUOTE (0 0 0)) (CADDR X))))
 			       (LIST (CAADDR X)
 				     (CADADR (CDR X))
-				     (DIFFERENCE 1000 (CADDR Y)))
+				     (DIFFERENCE 01000 (CADDR Y)))
 			       (CAR X))
 			(RETURN Y))))
     (THGOAL (#AT $?X $_W))
@@ -269,9 +269,9 @@
 	 (THCONSE (Y LOC)
 		  (#MOVEHAND2 $?Y)
 		  (COND ((EQUAL $?Y HANDAT) (THSUCCEED THEOREM))
-			((AND (LESSP 37 (CAR $?Y) 1141)
-			      (LESSP -1 (CADR $?Y) 1141)
-			      (LESSP -1 (CADDR $?Y) 1001))))
+			((AND (LESSP 037 (CAR $?Y) 01141)
+			      (LESSP -1 (CADR $?Y) 01141)
+			      (LESSP -1 (CADDR $?Y) 01001))))
 		  (THVSETQ $_LOC HANDAT)
 		  (THSETQ HANDAT $?Y)
 		  (THSETQ THTIME (ADD1 THTIME))
@@ -462,7 +462,7 @@
 	 (THCONSE ((WHY (EV)) EV)
 		  (#RAISEHAND)
 		  (MEMORY)
-		  (THGOAL (#MOVEHAND $E (LIST (CAR HANDAT) (CADR HANDAT) 1000))
+		  (THGOAL (#MOVEHAND $E (LIST (CAR HANDAT) (CADR HANDAT) 01000))
 			  (THNODB)
 			  (THUSE TC-MOVEHAND))
 		  (MEMOREND (#RAISEHAND $?EV)))
@@ -495,13 +495,13 @@
   (#STACKUP $?X)
   (OR (LESSP (APPLY (QUOTE PLUS)
 		    (MAPCAR (QUOTE (LAMBDA (X) (CADDR (SIZE X)))) $?X))
-	     1201)
-      (NOT (DPRINT2 (QUOTE TOO/ HIGH/,))))
+	     01201)
+      (NOT (DPRINT2 (QUOTE TOO\ HIGH\,))))
   (THCOND
    ((AND $?X (CDR $?X)))
    ((THSETQ
      $_X
-     (APPEND $?X
+     (APPEND (LISTIFY $?X)
 	     (THVAL (LIST (QUOTE THFIND)
 			  (COND ((NULL $?X) 3) (2))
 			  (QUOTE $?Y)
@@ -597,7 +597,7 @@
 (DEFPROP TCT-PICKUP
 	 (THCONSE (X EV TIME)
 		  (#PICKUP $?X $?TIME)
-		  (THOR (THAND (THGOAL (#PICKUP$?EV $?X)) (TIMECHK $?EV $?TIME))
+		  (THOR (THAND (THGOAL (#PICKUP $?EV $?X)) (TIMECHK $?EV $?TIME))
 			(THGOAL (#PICKUP $?EV $?X $?TIME) (THUSE TCTE-PICKUP))))
 	 THEOREM)
 
@@ -639,6 +639,15 @@
 		  (THAMONG $?X (LIST (CADDR $?Z))))
 	 THEOREM)
 
+(DEFPROP TCT-ON
+	 (THCONSE (Y X Z TIME)
+		  (#SUPPORT $?X $?Y $?TIME)
+		  (THOR (THGOAL (#MANIP $?Y)) (THGOAL (#IS $?Y #BOX)))
+		  (THAMONG $?Z (TFIND $?Y $?TIME))
+		  (NOT (LESSP (CAR $?Z) (OR (START? $?TIME) -1.)))
+		  (THAMONG $?X (LIST (CADDR $?Z))))
+	 THEOREM)
+
 (DEFPROP TCT-2
 	 (THCONSE (X EV TIME) ($?X $?TIME) (THGOAL ($?X $?EV $?TIME)
 						   (THUSE TCTE-3)))
@@ -686,7 +695,7 @@
 		  (TIMECHK $?EVENT $?TIME)
 		  (THOR (THGOAL (#PUTON $?EVENT $?X ?))
 			(THGOAL (#PICKUP $?EVENT $?X)))
-		  (OR (THVSETQ $_Z (SUB1 (ASSQ (GET $?EVENT (QUOTE END))
+		  (OR (THVSETQ $_Z (ASSQ (SUB1 (GET $?EVENT (QUOTE END))
 					       (GET $?X (QUOTE HISTORY)))))
 		      (ERT TCTE-PUT WRONG))
 		  (THAMONG $?Y (LIST (CADR $?Z)))
@@ -746,7 +755,7 @@
 (DEFPROP TE-EXISTS (THERASING (X) (#EXISTS $?X) (THSUCCEED)) THEOREM)
 
 (DEFPROP TE-SUPP
-	 (THERASING (X Y Z)
+	 (THERASING (X Y Z W)
 		    (#AT $?X ?)
 		    (THCOND ((THGOAL (#SUPPORT $?X $_Y)) (ERT TE-SUPP))
 			    ((THGOAL (#SUPPORT $_Y $?X))
